README.md

```diff
@@ -138,6 +141,27 @@ There are a couple of ways you can contribute to this repository:
 
 Before we can accept your pull request you will need to sign a **Contribution License Agreement**. All you need to do is to submit a pull request, then the PR will get appropriately labelled (for example `cla-required`, `cla-norequired`, `cla-signed`, `cla-already-signed`). If you already signed the agreement we will continue with reviewing the PR, otherwise system will tell you how you can sign the CLA. Once you sign the CLA all future PR's will be labeled as `cla-signed`.
 
+## Troubleshooting
+
+### I get "unauthorized: authentication required" in the terminal when executing some commands, such as "Docker: push".
+
+Make sure you are signed in to the Docker Hub or Azure container registry from the docker CLI via `docker login`.
+
+If you are using an Azure container registry, you will need to get the username and password from Azure by right-clicking on the Azure container registry in the extension and selecting "Browse in the Azure Portal", then selecting the "Access Keys" tab.
+![Getting Azure username and password](images/AzureUsernamePassword.png)
+
+Finally, execute `docker login`, for example:
+
+```bash
+docker login exampleazurecontainerregistry.azurecr.io
+```
+
+and respond with the username and password from Azure.
```

Comment:
Consider adding:  Note that the username is different from the email id. I've noticed some confusion arising from the docker website allowing one to use their email id in logging in. See [this docker issue](https://github.com/docker/hub-feedback/issues/831#issuecomment-259230362) for this happening.